modules/openapi-generator/src/main/java/org/openapitools/codegen/IJsonSchemaValidationProperties.java

```diff
@@ -213,4 +217,47 @@ default void setTypeProperties(Schema p) {
             setIsAnyType(true);
         }
     }
+
+    /**
+     * @return basic type - no generics supported.
+     */
+    default String getBaseType() {
+        return null;
+    };
+
+    /**
+     * @return complex type that can contain type parameters - like {@code List<Items>} for Java
+     */
+    default String getComplexType() {
+        return getBaseType();
+    };
+
+    /**
+     * Recursively collect all necessary imports to include so that the type may be resolved.
+     *
+     * @param includeContainerTypes whether or not to include the container types in the returned imports.
+     * @return all of the imports
+     */
+    default Set<String> getImports(boolean includeContainerTypes) {
+        Set<String> imports = new HashSet<>();
+        if (includeContainerTypes || !(this.getIsArray() || this.getIsMap())) {
+            if (this.getComposedSchemas() != null) {
+                CodegenComposedSchemas composed = (CodegenComposedSchemas) this.getComposedSchemas();
+                List<CodegenProperty> allOfs =  composed.getAllOf() == null ? Collections.emptyList() : composed.getAllOf();
+                List<CodegenProperty> oneOfs =  composed.getOneOf() == null ? Collections.emptyList() : composed.getOneOf();
+                Stream<CodegenProperty> innerTypes = Stream.concat(allOfs.stream(), oneOfs.stream());
+                innerTypes.flatMap(cp -> cp.getImports(includeContainerTypes).stream()).forEach(s -> imports.add(s));
+            } else {
+                imports.add(this.getComplexType());
+                imports.add(this.getBaseType());
+            }
+        }
+        if (this.getItems() !=null) {
```

Comment:
DefaultCodegen does a horrible thing where it assigns properties into items. Can you also limit this to: `this.getItems() !=null && this.getIsArray`?

Reply:
Can you elaborate why this is needed?

Reply:
Thanks!